src/EasyAbp.Abp.PhoneNumberLogin.Application/EasyAbp/Abp/PhoneNumberLogin/Account/PhoneNumberLoginAccountAppService.cs

```diff
@@ -1,56 +1,274 @@
+
+            var identityUser = await _uniquePhoneNumberIdentityUserRepository.GetByConfirmedPhoneNumberAsync(input.PhoneNumber);
+
+            (await _identityUserManager.ResetPasswordAsync(identityUser, input.VerificationCode, input.Password)).CheckErrors();
         }
 
-        public virtual Task<string> RequestTokenByVerificationCodeAsync(RequestTokenByVerificationCodeInput input)
+        public virtual async Task<TryRegisterAndRequestTokenResult> TryRegisterAndRequestTokenAsync(LoginInput input)
         {
-            throw new System.NotImplementedException();
+            var result = await GetValidateResultAsync(input.PhoneNumber, input.VerificationCode, VerificationCodeType.Register);
+
+            if (!result)
+            {
+                throw new InvalidVerificationCodeException();
+            }
+
+            await _identityOptions.SetAsync();
+
+            bool registerUser = false;
+
+            var identityUser =
+                await _uniquePhoneNumberIdentityUserRepository.FindByConfirmedPhoneNumberAsync(input.PhoneNumber);
+
+            if (identityUser is null)
+            {
+                using (var uow = UnitOfWorkManager.Begin(new AbpUnitOfWorkOptions(true), true))
+                {
+                    identityUser = await _phoneNumberLoginNewUserCreator.CreateAsync(input.PhoneNumber, input.UserName, input.Password, input.EmailAddress);
+
+                    await uow.CompleteAsync();
+                }
+
+                registerUser = true;
+            }
+
+            string code = await GenerateCodeAsync(input.PhoneNumber, VerificationCodeType.Login, identityUser);
+
+            return new TryRegisterAndRequestTokenResult(
+                registerUser ? LoginResultType.Register : LoginResultType.Login,
+                (await RequestIds4LoginByCodeAsync(input.PhoneNumber, code))?.Raw,
+                CurrentTenant.Id);
+        }
+
+        public virtual async Task<string> RequestTokenByPasswordAsync(RequestTokenByPasswordInput input)
+        {
+            return (await RequestIds4LoginByPasswordAsync(input.PhoneNumber, input.Password))?.Raw;
+        }
+
+        public virtual async Task<string> RequestTokenByVerificationCodeAsync(RequestTokenByVerificationCodeInput input)
+        {
+            return (await RequestIds4LoginByCodeAsync(input.PhoneNumber, input.VerificationCode))?.Raw;
         }
 
         public virtual async Task<string> RefreshTokenAsync(RefreshTokenInput input)
         {
             return (await RequestIds4RefreshAsync(input.RefreshToken))?.Raw;
         }
-        
-        protected virtual Task<TokenResponse> RequestIds4LoginAsync(string appId, string unionId, string openId)
+
+        protected virtual async Task<bool> GetValidateResultAsync(string phoneNumber, string code, VerificationCodeType type)
         {
-            throw new System.NotImplementedException();
+            return await _verificationCodeManager.ValidateAsync(
```

Comment:
Similar to the `GenerateCodeAsync` method, `GetValidateResultAsync` should provide validation for all type of verification code.